### PR TITLE
Fix docs and test

### PR DESCRIPTION
**Pull request by thomaspinder**
This PR fixes some bugs within the documentation. The bugs are all hangovers from the recent update to Jax v0.4. 

The PR also includes a testing job to our CircleCI workflow to ensure that all notebooks compile before using them to build the documentation site.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The docs do not currently build. Worse still, we will not know this as they are not being tested.

Issue Number: #169  #170

### CODE REVIEW

File: examples/haiku.pct.py [hunk 1]
```diff
@@ -106,12 +105,12 @@ def __call__(
...
 
-    def initialise(self, dummy_x: Float[Array, "1 D"], key: PRNGKeyType) -> None:
+    def initialise(self, dummy_x: Float[Array, "1 D"], key: jr.PRNGKey) -> None:
         nn_params = self.network.init(rng=key, x=dummy_x)
         base_kernel_params = self.base_kernel._initialise_params(key)
         self._params = {**nn_params, **base_kernel_params}
 
-    def _initialise_params(self, key: PRNGKeyType) -> Dict:
+    def _initialise_params(self, key: jr.PRNGKey) -> Dict:
         return self._params
```
Comment (Daniel-Dodd) [inline examples/haiku.pct.py:114]: `jr.KeyArray`?

Comment (Daniel-Dodd) [reply examples/haiku.pct.py:114]: (instead of `jr.PRNGKey`)

**Comment by thomaspinder** [comment]
@Daniel-Dodd I'm going to move the docs building back to a Github workflow for now.

**Review by Daniel-Dodd — APPROVED** [review]
Super @thomaspinder, this looks good to me.

**Pull request merged**